chore: move tab component to storybook

jakobsitory opened the pull request:
<!-- We require pull request titles to follow the Conventional Commits specification ( https://www.conventionalcommits.org/en/v1.0.0/#summary ). Please make sure your title follow these conventions -->

## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes https://github.com/formbricks/formbricks-com/issues/138



## How should this be tested?


- `pnpm go storybook`
- tabnav: http://localhost:6006/?path=/docs/ui-tabnav--docs
- Tabs: http://localhost:6006/?path=/docs/ui-tabs--docs

## Checklist

<!-- We're starting to get more and more contributions. Please help us making this efficient for all of us and go through this checklist. Please tick off what you did  -->

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [X] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

jobenjada commented:
hey!

The tabsbar looks good ✅ 

### Tabs

1. Fix width
Imo we should make sure that the total width of the tabs component doesn't change when we switch between options (it also doesn't in [shadcn](https://ui.shadcn.com/docs/components/tabs)). Might be due to font weights or so, pls align with shadcn:

<img width="214" height="65" alt="image" src="https://github.com/user-attachments/assets/97fa8931-7b09-4a19-bf28-8498444e2b4b" />

Big size also has this issue:
<img width="268" height="160" alt="image" src="https://github.com/user-attachments/assets/6ff7409d-41e9-4fc2-a9fb-ce9c25a50c16" />


Funnily, it doesn't happen with Icons:
<img width="368" height="146" alt="image" src="https://github.com/user-attachments/assets/f9acb3e2-8af4-4b01-8de4-9af03eda019e" />


2. Responsiveness: What happens if the width of all of the tabs does not fit the max width of the parent container? We should design for that case as it might happen e.g. in the survey editor or in modals:
<img width="374" height="126" alt="image" src="https://github.com/user-attachments/assets/c54404f2-a5ba-4a13-b457-6abbb48d6bd6" />

also, the icons are scaling inconsistently:
<img width="323" height="73" alt="image" src="https://github.com/user-attachments/assets/81e33df2-eb05-44b7-99c1-7a2049e1ec1c" />

Maybe "big size" is an option to save width (icon above and small text)

the rest looks good :)

jobenjada reviewed (changes requested):
see above

jakobsitory commented:
- Updated the component, the visual jump on interacrion was due to the changing width of the content below the the tabs. 
- Introduced a fit and fill variabt to handle all use cases on Formbricks 
- Introduced overflow-x, so tabs can be scrolled horizontakly if theg exceed the available width. (that should only be the edge case, but nothing breaks anymore)
- Couldn't replicate the icon sizing issue. Likely due to the other changes I made

jakobsitory commented on apps/web/modules/ui/components/tab-nav/index.tsx:
```diff
@@ -0,0 +1,53 @@
+  activeTabClassName?: string;
+}
+
+const Nav: React.FC<NavProps> = ({ tabs, activeId, setActiveId, activeTabClassName }) => {
+  return (
+    <nav className="flex h-full items-center space-x-3" aria-label="Tabs">
+      {tabs.map((tab) => (
+        <button
+          key={tab.id}
+          onClick={() => setActiveId(tab.id)}
+          className={cn(
+            tab.id === activeId
+              ? `border-brand-dark text-primary border-b-2 font-semibold ${activeTabClassName}`
+              : "text-slate-500 hover:text-slate-700",
+            "flex h-full items-center px-3 text-sm font-medium"
+          )}
+          aria-current={tab.id === activeId ? "page" : undefined}>
+          {tab.icon && <div className="flex h-5 w-5 items-center">{tab.icon}</div>}
+          {tab.label}
+        </button>
+      ))}
+    </nav>
+  );
+};
```
Fixed in next commit
